python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -1998,10 +1998,19 @@ def get(
     def prepare_get_for_external_client(self, ref: DatasetRef) -> FileDatastoreGetPayload:
         # Docstring inherited
 
+        # 8 hours.  Chosen somewhat arbitrarily -- this is long enough that the
```

Comment:
It won't cut off the download, but if you need to retry at the HTTP layer it will need the signature to remain valid.  Its nice if you can allow HTTP retries without needing to throw an exception all the way up to Butler.get() to re-do the API request too.

Or e.g. for multiple range requests if it does some slow work in between reading the chunks, the signature will need to stay valid for the duration of the work.

Also we had discussed like a get_many(), so the download might not start immediately if there are other files in queue on a slow link.

I would argue that a few minutes isn't really long enough.  10-20 minutes is probably OK, but I think it's best if we can make this the longest possible length that isn't a political problem.  How would you feel about 1-2 hours? I think it's best to have timeouts one or two orders of magnitude higher than the worst expected typical case, so things don't blow up immediately when the system is under stress.

Also I really don't buy the social media thing given that we provide a function for downloading the files, which could then be stored wherever you want for as long as you want.